### PR TITLE
Update CI scripts

### LINKED ISSUE #60: [Flake] ./hack/.ci/run-e2e-gke.sh: line 67: exit: : numeric argument required
```
 ++ unblock-must-gather-pod
++ kubectl -n e2e exec pod/must-gather -- bash -euEo pipefail -O inherit_errexit -c 'touch /tmp/exit'
+ [[ -f /tmp/exit ]]
+ sleep 1
+ [[ -f /tmp/exit ]]
++ kubectl -n e2e get pods/must-gather '--output=jsonpath={.status.containerStatuses[0].state.terminated.exitCode}'
+ exit_code=
+ kubectl -n e2e delete pod/must-gather --wait=false
pod "must-gather" deleted
+ [[ '' != \0 ]]
+ echo 'Collecting artifacts using must-gather failed'
Collecting artifacts using must-gather failed
+ exit ''
./hack/.ci/run-e2e-gke.sh: line 67: exit: : numeric argument required 
```

Flakes on master several times in a row, e.g. https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_local-csi-driver/57/pull-local-csi-driver-master-e2e-gke-parallel/1813527225367007232#1:test-build-log.txt%3A5892

### PR DESCRIPTION
**Pull request by tnozicka**
This PR updates CI scripts based on the latest changes in scylla-operator - where they come from (with some modifications).

Fixes #60

### CODE REVIEW

File: hack/.ci/lib/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#
+# Copyright (C) 2024 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+source "$( dirname "${BASH_SOURCE[0]}" )/../../lib/kube.sh"
+
+SO_IMAGE="${SO_IMAGE:-quay.io/scylladb/scylla-operator:latest}"
```
Comment (rzetelskik) [inline hack/.ci/lib/e2e.sh:11]: shouldn't it be pinned actually?

Comment (tnozicka) [reply hack/.ci/lib/e2e.sh:11]: it's the same as it was
https://github.com/scylladb/local-csi-driver/pull/61/files#diff-2eb61661d79cabc653bae4339195fa6c9aeb8ddfd3e0b701c223a0334c6369d3L45

For a master branch, `latest` seems like a reasonable choice but it's a tradeoff one way or the other.

**Comment by rzetelskik** [comment]
lgtm
/assign zimnx

**Review by zimnx — APPROVED** [review]
/lgtm

**Pull request merged**